xapian-applications/omega/handler_gmime.cc

```diff
@@ -54,8 +55,8 @@ extract_html(const string& text, string& charset, string& dump)
 size_t decode(unsigned char* text, size_t len, GMimeContentEncoding encoding)
 {
     unsigned char buffer[SIZE];
-    int state;
-    guint32 save;
+    static int state = 0;
+    static guint32 save = 0;
```

Comment:
Making these `static` isn't really a good solution.  Then they won't get reset between mime entities so if one leaves the decoder in a different state to (0,0) then the next will start from that state (and potentially in could be the state from a different decoder even).  It's also not thread safe, though we aren't currently using threads here.

Instead declare them in the caller and pass them in by reference - then they have the lifetime we actually want.

Reply:
Okay, I am going to change it.